test/dtypes.c

```diff
@@ -8954,8 +8954,14 @@ test_encode(void)
     unsigned char *vlstr_buf      = NULL;
     hid_t          ret_id;
     herr_t         ret;
+    char           test_msg[128];
 
-    TESTING("functions of encoding and decoding datatypes");
+    /* Silence unused parameter warning if built with no deprecated symbols */
+    (void)use_old_decode_api;
```

Comment:
Instead of silencing this way, using the `H5_ATTR_DEPRECATED_USED` macro would be a little cleaner